### PR TITLE
get network area diagram

### PR DESCRIPTION
**Pull request by etiennehomer**
Signed-off-by: Etienne Homer <etiennehomer@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature: getting Network Area Diagram SVG

**What is the current behavior?** *(You can also link to an open issue here)*
NAD genreration is not implemented.

**What is the new behavior (if this is a feature change)?**
We have an endpoint to get the svg of a network area.

### CODE REVIEW

File: src/main/java/com/powsybl/sld/server/NetworkAreaDiagramService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * @author Etienne Homer<etienne.homer at rte-france.com>
+ */
+@ComponentScan(basePackageClasses = {NetworkStoreService.class})
+@Service
+class NetworkAreaDiagramService {
+    @Autowired
+    private NetworkStoreService networkStoreService;
+
+    public String generateNetworkAreaDiagramSvg(UUID networkUuid, String variantId, List<String> voltageLevelsIds, int depth) {
+        Network network = SingleLineDiagramService.getNetwork(networkUuid, variantId, networkStoreService, PreloadingStrategy.COLLECTION);
```
Comment (geofjamg) [inline src/main/java/com/powsybl/sld/server/NetworkAreaDiagramService.java:36]: To move in a utility class?

Comment (etiennehomer) [reply src/main/java/com/powsybl/sld/server/NetworkAreaDiagramService.java]: done

File: src/main/java/com/powsybl/sld/server/DiagramUtils.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.powsybl.sld.server;
```
Comment (geofjamg) [inline src/main/java/com/powsybl/sld/server/DiagramUtils.java:1]: license

File: src/main/java/com/powsybl/sld/server/DiagramUtils.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.network.store.client.NetworkStoreService;
+import com.powsybl.network.store.client.PreloadingStrategy;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.UUID;
+
+public final class DiagramUtils {
```
Comment (geofjamg) [inline src/main/java/com/powsybl/sld/server/DiagramUtils.java:12]: author

Comment (etiennehomer) [reply src/main/java/com/powsybl/sld/server/DiagramUtils.java]: done

File: src/main/java/com/powsybl/sld/server/DiagramUtils.java (hunk 2 above)
Comment (etiennehomer) [reply src/main/java/com/powsybl/sld/server/DiagramUtils.java]: done

**Review by geofjamg — APPROVED** [review]

**Pull request merged**